libraries/plugins/peerplays_sidechain/sidechain_net_handler_bitcoin.cpp

```diff
@@ -852,86 +894,105 @@ void sidechain_net_handler_bitcoin::recreate_primary_wallet() {
 
          auto active_sons = gpo.active_sons;
          vector<string> son_pubkeys_bitcoin;
+         vector<uint64_t> son_votes;
          for (const son_info &si : active_sons) {
             son_pubkeys_bitcoin.push_back(si.sidechain_public_keys.at(sidechain_type::bitcoin));
+            son_votes.push_back(si.weight);
          }
 
-         if (!wallet_password.empty()) {
-            bitcoin_client->walletpassphrase(wallet_password, 5);
+         std::string full_address_info = create_multisig_address(son_pubkeys_bitcoin, son_votes);
```

Comment:
The only change in this function should be the call to address creation function, and eventually call to transaction creation. Looks like most of existing changes with fixes are reverted. There are two calls to create_transaction. Refactor this function, to match the one from current SONs-base.

```
          auto active_sons = gpo.active_sons;
          vector<string> son_pubkeys_bitcoin;
+         vector<uint64_t> son_weights;
          for (const son_info &si : active_sons) {
             son_pubkeys_bitcoin.push_back(si.sidechain_public_keys.at(sidechain_type::bitcoin));
+            son_weights.push_back(si.weight);
          }
 
-         if (!wallet_password.empty()) {
-            bitcoin_client->walletpassphrase(wallet_password, 5);
-         }
-         uint32_t nrequired = son_pubkeys_bitcoin.size() * 2 / 3 + 1;
-         string reply_str = bitcoin_client->addmultisigaddress(nrequired, son_pubkeys_bitcoin);
+         string address_info = create_multisig_address(son_pubkeys_bitcoin, son_weights);
 
-         std::stringstream active_pw_ss(reply_str);
+         std::stringstream active_pw_ss(address_info);
          boost::property_tree::ptree active_pw_pt;
          boost::property_tree::read_json(active_pw_ss, active_pw_pt);
          if (active_pw_pt.count("error") && active_pw_pt.get_child("error").empty()) {
```

Reply:
Working with walletm  calculating nrequired and parsing jsob strings are psbt-specific actions,  so I've moved them to psbt-specific calls.